packages/material-ui/src/TablePagination/TablePagination.js

```diff
@@ -1,75 +1,135 @@
+  Toolbar,
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Toolbar',
+    overridesResolver: (props, styles) => styles.toolbar,
   },
-  /* Styles applied to the Toolbar component. */
-  toolbar: {
+)(({ theme }) => ({
+  minHeight: 52,
+  paddingRight: 2,
+  [`${theme.breakpoints.up('xs')} and (orientation: landscape)`]: {
     minHeight: 52,
-    paddingRight: 2,
   },
-  /* Styles applied to the spacer element. */
-  spacer: {
-    flex: '1 1 100%',
+  [theme.breakpoints.up('sm')]: {
+    minHeight: 52,
+    paddingRight: 2,
   },
-  /* Styles applied to the select label Typography element. */
-  selectLabel: {
+  [`& .${tablePaginationClasses.actions}`]: {
     flexShrink: 0,
+    marginLeft: 20,
+  },
+}));
+
+const TablePaginationSpacer = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Spacer',
+    overridesResolver: (props, styles) => styles.spacer,
   },
-  // TODO v5: `.selectRoot` should be merged with `.input`
-  /* Styles applied to the Select component root element. */
-  selectRoot: {
-    marginRight: 32,
-    marginLeft: 8,
+)({
+  flex: '1 1 100%',
+});
+
+const TablePaginationSelectLabel = experimentalStyled(
+  'p',
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'SelectLabel',
+    overridesResolver: (props, styles) => styles.selectLabel,
+  },
+)(({ theme }) => ({
+  ...theme.typography.body2,
+  flexShrink: 0,
+}));
+
+const TablePaginationSelect = experimentalStyled(
+  Select,
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Select',
+    overridesResolver: (props, styles) => ({
+      [`& .${tablePaginationClasses.selectIcon}`]: styles.selectIcon,
+      [`& .${tablePaginationClasses.select}`]: styles.select,
+      ...styles.input,
+      ...styles.selectRoot,
+    }),
   },
-  /* Styles applied to the Select component `select` class. */
-  select: {
+)({
+  color: 'inherit',
+  fontSize: 'inherit',
+  flexShrink: 0,
+  marginRight: 32,
+  marginLeft: 8,
+  [`& .${tablePaginationClasses.input}`]: {
     paddingLeft: 8,
     paddingRight: 24,
     textAlign: 'right',
     textAlignLast: 'right', // Align <select> on Chrome.
   },
-  // TODO v5: remove
-  /* Styles applied to the Select component `icon` class. */
-  selectIcon: {},
-  /* Styles applied to the InputBase component. */
-  input: {
-    color: 'inherit',
-    fontSize: 'inherit',
-    flexShrink: 0,
-  },
-  /* Styles applied to the MenuItem component. */
-  menuItem: {},
-  /* Styles applied to the displayed rows Typography element. */
-  displayedRows: {
-    flexShrink: 0,
+});
+
+const TablePaginationMenuItem = experimentalStyled(
+  MenuItem,
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'MenuItem',
+    overridesResolver: (props, styles) => styles.menuItem,
   },
-  /* Styles applied to the internal `TablePaginationActions` component. */
-  actions: {
-    flexShrink: 0,
-    marginLeft: 20,
+)();
+
+const TablePaginationDisplayedRows = experimentalStyled(
+  'p',
```

Comment:
```suggestion
  Typography,
```

Wasn't this previously `Typography`?

Reply:
It was, it removed it for simplicity.

Reply:
Got it, ok I thought it was overlooked